library/include/borealis/core/touch/pan_gesture_recognizer.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+    Copyright 2021 XITRIX
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#pragma once
+
+#include "gesture_recognizer.hpp"
+
+namespace brls
+{
+
+class PanGestureRecognizer;
+typedef std::function<void(PanGestureRecognizer*)> PanGestureRespond;
```

Comment:
Do you think it would be possible to change the signature of the listener to have everything passed as an individual parameter instead of giving the huge recognizer? Like `typedef std::function<void(float y, float acceleration...)> PanGestureRespond;`

Or if it's too boilerplate, maybe create a struct "PanGestureState" that holds everything and give a pointer to that instead?

Reply:
I'm not sure about changing it. I think having full recogniser in callback will give us more possibilities to control it, i.e. may be we'd like to disable it when it's done. Also I think having a separate struct in callback is just duplication of already existing data, it's much easier to make some fields in recogniser as public and allow to take them. What do you think about it?

Reply:
I feel like if you need the recognizer in the callback you should capture it in the lambda. The callback is for "state change" event and so it should only give new state data...

Speaking of which, would it be possible to use brls::Event instead of a "raw" std::function?

Reply:
I changed whole recognizer to status structs.
About brls::Event, I don't think that it's what we should use with recognizers, I feel like recognizer should call one and only one function which could include invoking Events inside it, but not contain an event subscriber in recognizer it self. I can't think of a single example where having several callbacks on recognizer could be useful. Anyway, if you think it should have this functionality, that's not a problem, I'll add it.